apis/python/src/tiledbsoma/_query.py

```diff
@@ -524,23 +524,141 @@ def to_spatialdata(  # type: ignore[no-untyped-def]
                 Defaults to ``obs``.
         """
 
-        from spatialdata import SpatialData
-
-        from .io.spatial.outgest import _add_scene_to_spatialdata
+        import spatialdata as sd
+
+        from ._multiscale_image import MultiscaleImage
+        from ._point_cloud_dataframe import PointCloudDataFrame
+        from .io.spatial.outgest import (
+            _convert_axis_names,
+            _get_transform_from_collection,
+            to_spatialdata_image,
+            to_spatialdata_multiscale_image,
+            to_spatialdata_points,
+            to_spatialdata_shapes,
+        )
 
         warnings.warn(SPATIAL_DISCLAIMER)
 
         # Get a list of scenes to add to SpatialData object.
         if scene_presence_mode == "obs":
-            scene_ids = self.obs_scene_ids()
+            scene_names = tuple(str(scene_name) for scene_name in self.obs_scene_ids())
         elif scene_presence_mode == "var":
-            scene_ids = self.var_scene_ids()
+            scene_names = tuple(str(scene_name) for scene_name in self.var_scene_ids())
         else:
             raise ValueError(
                 f"Invalid scene presence mode '{scene_presence_mode}'. Valid options "
                 f"are 'obs' and 'var'."
             )
 
+        # Create empty SpatialData instance and dict to store region/instance keys.
+        sdata = sd.SpatialData()
+        region_joinids: Dict[str, Any] = {}
+
+        # Add data from linked scenes.
+        for scene_name in scene_names:
+            scene = self.experiment.spatial[scene_name]
+
+            # Cannot have spatial data if no coordinate space.
+            if scene.coordinate_space is None:
+                continue
+
+            # Get the map from Scene dimension names to SpatialData dimension names.
+            input_axis_names = scene.coordinate_space.axis_names
+            _, scene_dim_map = _convert_axis_names(input_axis_names, input_axis_names)
+
+            # Export obsl data to SpatialData.
+            if "obsl" in scene:
```

Comment:
Not a priority now but in the future might be nice to extract this whole obsl/varl block in a single helper method that can handle both axes.